File: examples/node/src/discount.js
export function applyDiscountCents(priceCents, rateBps) {
  if (!Number.isInteger(priceCents) || priceCents < 0) {
    throw new RangeError('priceCents must be an integer >= 0');
  }
  if (!Number.isInteger(rateBps) || rateBps < 0 || rateBps > 10_000) {
    throw new RangeError('rateBps must be an integer between 0 and 10_000');
  }

  const discountedScaled = priceCents * (10_000 - rateBps);
  return Math.floor(discountedScaled / 10_000);
}

File: examples/python/discount.py
def apply_discount_cents(price_cents: int, rate_bps: int) -> int:
    if price_cents < 0:
        raise ValueError("price_cents must be >= 0")
    if rate_bps < 0 or rate_bps > 10_000:
        raise ValueError("rate_bps must be between 0 and 10_000")

    discounted_scaled = price_cents * (10_000 - rate_bps)
    return (discounted_scaled + 5_000) // 10_000

